that_depends/providers/collections.py

```diff
@@ -18,3 +18,26 @@ def sync_resolve(self) -> list[T]:
 
     async def __call__(self) -> list[T]:
         return await self.async_resolve()
+
+
+class Selector(AbstractProvider[T]):
```

Comment:
It's better to move it to separate package `selector`.
Collections is for lists and dicts